### PR TITLE
Add pytest to github actions

### LINKED ISSUE #210: Fix CI
Since June 15th, 2021, the building on [travis-ci.org](http://www.travis-ci.org/) is ceased. Please use [travis-ci.com](http://www.travis-ci.com/) from now on.

### PR DESCRIPTION
**Pull request by cverstege**
Depending on the python version 1-2 unittests are failing currently. This needs investigation.
But in general the pytest command runs successfully. I'm open for improvements

### CODE REVIEW

**Comment by cverstege** [comment]
Closes #210

File: .github/workflows/pytest.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.7", "3.8", "3.9", "3.10", "3.11"]
+
+    steps:
+      - uses: actions/checkout@v3
```
Comment (JohannesGaessler) [inline .github/workflows/pytest.yml:14]: I don't understand why you are using v3 here and v4 later but evidently it works.

**Review by JohannesGaessler — APPROVED** [review]
As far as I can tell this works; I'm aware of the failing unittest due to matplotlib printing a warning but I did not yet get around to fixing it.

File: .github/workflows/pytest.yml (hunk 1 above)
Comment (cverstege) [reply .github/workflows/pytest.yml:14]: That's from the example provided by GitHub https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python

**Comment by cverstege** [comment]
I think we should fix the failing unit test before merging this. It will look bad if the checks on the main branch are failing.

**Pull request merged**